### PR TITLE
allow trailing condition discriminator

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

File: test/test_CatsParser.py [hunk 1]
```diff
@@ -195,7 +206,31 @@ def test_can_parse_struct_conditional_types(self):
         self.assertEqual(type_descriptors['Enclosing'], {'type': 'struct', 'comments': '', 'layout': [
             {'name': 'enclosingType', 'type': 'Shape', 'comments': ''},
             {'name': 'circumference', 'type': 'Circ', 'condition': 'enclosingType', 'condition_value': 'circle', 'comments': 'u part 1'},
-            {'name': 'perimiter', 'type': 'Perm', 'condition': 'enclosingType', 'condition_value': 'rectangle', 'comments': 'union part 2'}
+            {'name': 'perimiter', 'type': 'Perm', 'condition': 'enclosingType', 'condition_value': 'rectangle', 'comments': 'union pt 2'}
+        ]})
+
+    def test_can_parse_struct_conditional_types_reverse(self):
```
Comment (Jaguar0625) [inline test/test_CatsParser.py]: suggestion for better test name or current ok ?

File: test/test_StructParser.py [hunk 2]
```diff
@@ -80,6 +80,22 @@ def test_can_append_scalar_conditional(self):
             {'name': 'bar', 'condition': 'foo', 'condition_value': 'red'}
         ]}), result)
 
+    def test_can_append_scalar_conditional_trailing_reference(self):
```
Comment (Jaguar0625) [inline test/test_StructParser.py]: test name ok?

File: test/test_CatsParser.py [hunk 3]
```diff
@@ -186,7 +197,7 @@ def test_can_parse_struct_conditional_types(self):
             '\tenclosingType = Shape',
             '\t# u part 1',
             '\tcircumference = Circ if enclosingType equals circle',
-            '\t# union part 2',
+            '\t# union pt 2',
```
Comment (gimer) [inline test/test_CatsParser.py]: ?

File: test/test_CatsParser.py (hunk 1 above)
Comment (gimer) [reply test/test_CatsParser.py]: types_inversed, types_out_of_order? not sure

File: test/test_CatsParser.py (hunk 3 above)
Comment (Jaguar0625) [reply test/test_CatsParser.py]: so same strings could be used in normal and reverse test variants without line wrapping

**Pull request merged**